server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -835,6 +852,41 @@ public String toString() {
         }
     }
 
+    private static class RelativeTimeSupplier implements TimeSupplier {
+        private final CachedTimeThread cachedTimeThread;
+
+        private RelativeTimeSupplier(CachedTimeThread cachedTimeThread) {
+            this.cachedTimeThread = cachedTimeThread;
+        }
+
+        @Override
+        public long timeInMillis() {
+            return cachedTimeThread.relativeTimeInMillis();
+        }
+
+        @Override
+        public long timeInNanos() {
+            return cachedTimeThread.relativeTimeInNanos();
+        }
+
+        @Override
+        public String toString() {
+            return ThreadPool.class.getCanonicalName() + "::relativeTimeInMillis";
+        }
```

Comment:
:+1: lets also have toString on RawRelativeTimeSupplier

Reply:
I'm leaving these kind of details for the final round once we agree what to do. I'll move this to draft.